src/scripting/bind.rs

```diff
@@ -105,6 +104,74 @@ fn to_scylla_value(v: &Value, typ: &ColumnType) -> Result<CqlValue, CassError> {
             let cql_time = CqlTime::try_from(naive_time)?;
             Ok(CqlValue::Time(cql_time))
         }
+        (Value::String(s), ColumnType::Native(NativeType::Duration)) => {
+            // TODO: add support for the following 'ISO 8601' format variants:
+            // - ISO 8601 format: P[n]Y[n]M[n]DT[n]H[n]M[n]S or P[n]W
+            // - ISO 8601 alternative format: P[YYYY]-[MM]-[DD]T[hh]:[mm]:[ss]
+            // See: https://opensource.docs.scylladb.com/stable/cql/types.html#working-with-durations
+            let duration_str = s.borrow_ref().unwrap();
+            if duration_str.is_empty() {
+                return Err(CassError(CassErrorKind::QueryParamConversion(
+                    format!("{:?}", v),
+                    "NativeType::Duration".to_string(),
+                    Some("Duration cannot be empty".to_string()),
+                )));
+            }
+            // NOTE: we parse the duration explicitly because of the 'CqlDuration' type specifics.
+            // It stores only months, days and nanoseconds.
+            // So, we do not translate days to months and hours to days because those are ambiguous
+            let re = Regex::new(concat!(
```

Comment:
Is this code run frequently? if so, let's compile it once and reuse it

Reply:
Indeed! Good catch!